### PR TITLE
fix: [IABT-1456] Update lollipop migration

### PR DESCRIPTION
**Pull request by hevelius**
## Short description
This PR resolves an issue that cause the crash on iOS and Android related to the migration of the lollipop persisted state

## List of changes proposed in this pull request
- Update MigrationManifest in `index.ts`
- Add unit test

## How to test
Test should solves correctly

### CODE REVIEW

File: ts/features/lollipop/store/index.ts [hunk 1]
```diff
@@ -8,12 +8,62 @@ import {
   PersistPartial,
   persistReducer
 } from "redux-persist";
+import { pipe } from "fp-ts/lib/function";
 import { Action } from "../../../store/actions/types";
 import { isDevEnv } from "../../../utils/environment";
 import lollipopReducer, { LollipopState } from "./reducers/lollipop";
 
 export const CURRENT_REDUX_LOLLIPOP_STORE_VERSION = 0;
```
Comment (shadowsheep1) [inline ts/features/lollipop/store/index.ts:16]: This should be set to `1`. Ain't it?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/features/lollipop/store/index.ts (hunk 1 above)
Comment (shadowsheep1) [inline ts/features/lollipop/store/index.ts:16]:
```suggestion
export const CURRENT_REDUX_LOLLIPOP_STORE_VERSION = 1;
```

File: ts/features/lollipop/store/index.ts [hunk 2]
```diff
@@ -8,11 +8,61 @@ import {
...
 } from "redux-persist";
+import { pipe } from "fp-ts/lib/function";
 import { Action } from "../../../store/actions/types";
 import { isDevEnv } from "../../../utils/environment";
 import lollipopReducer, { LollipopState } from "./reducers/lollipop";
 
-export const CURRENT_REDUX_LOLLIPOP_STORE_VERSION = 0;
+export const CURRENT_REDUX_LOLLIPOP_STORE_VERSION = 1;
+
+export const migrationKeyTag = (
```
Comment (CrisTofani) [inline ts/features/lollipop/store/index.ts:18]: Apart from tests this function seems not to be used anywhere, do we actually need it?

**Review by shadowsheep1 — COMMENTED** [review]
LGTM using `migrationKeyTagFunctional`

File: ts/features/lollipop/store/index.ts (hunk 2 above)
Comment (CrisTofani) [reply ts/features/lollipop/store/index.ts:18]: let's move it to test file then 😉

**Review by CrisTofani — COMMENTED** [review]
apart from changes suggested LGTM the PR works

**Review by Vangaorth — APPROVED** [review]
LGTM

**Review by CrisTofani — APPROVED** [review]

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**